src/Spectre.libGaussianMixtureModelling/MaximizationRunnerRef.h

```diff
@@ -48,81 +48,114 @@ limitations under the License.
 namespace spectre::unsupervised::gmm
 {
 /// <summary>
-/// Class serves the purpose of maximization step of Expectation Maximization algorithm. 
-/// Serves as a reference to learn from and is purpously not optimized to avoid 
-/// obfuscation of the code.
+/// Class serves the purpose of maximization step of Expectation Maximization
+/// algorithm. Serves as a reference to learn from and is purpously not
+/// optimized to avoid obfuscation of the code.
 /// </summary>
 class MaximizationRunnerRef
 {
 public:
     /// <summary>
-    /// Constructor initializing the class with data required during maximization step.
+    /// Constructor initializing the class with data required during
+    /// maximization step.
     /// </summary>
     /// <param name="spectrum">Spectrum to base calculations on.</param>
-    /// <param name="affilationMatrix">Matrix symbolising the probability of affilation
-    /// of each sample to a certain gaussian component.</param>
+    MaximizationRunnerRef(SpectrumView spectrum)
+    {
+        spectrum.mzs.size(); // Needed to avoid warning of unused param.
+    }
+
+    /// <summary>
+    /// Updates weights, means and tandard deviations in a single run.
+    /// </summary>
+    /// <param name="affilationMatrix">Matrix symbolising the probability
+    /// of affilation of each sample to a certain gaussian component.
+    /// </param>
+    /// <param name="spectrum">Spectrum to base calculations on.</param>
     /// <param name="components">Gaussian components to be updated</param>
-    /// <exception cref="ArgumentNullException">Thrown when either of mzArray or intensities pointers are null</exception>
-    MaximizationRunnerRef(SpectrumView spectrum, Matrix<DataType> &affilationMatrix, std::vector<GaussianComponent> &components)
-        : m_Spectrum(spectrum), m_AffilationMatrix(affilationMatrix), m_Components(components)
+    void Maximization(Matrix<DataType> &affilationMatrix,
+        SpectrumView spectrum, GaussianMixtureModel& components)
     {
```

Comment:
Why do we have definitions in `.h` file?

Reply:
That's how the previous API used to look like. It limits the number of translation units and allows compiler for doing better optimizations.

Reply:
This requires recompilation in each dependent project, while step itself will be used in one context: context of GMM. I'd understand such approach with your argument for GMM itself, but not for its elements. Maybe you could clarify it a bit more?